x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/connector_configuration_form.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiForm,
+  EuiFormRow,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiButtonEmpty,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { Status } from '../../../../../../common/types/api';
+
+import { ConnectorConfigurationApiLogic } from '../../../api/connector_package/update_connector_configuration_api_logic';
+
+import { ConnectorConfigurationLogic } from './connector_configuration_logic';
+
+export const ConnectorConfigurationForm = () => {
+  const { status } = useValues(ConnectorConfigurationApiLogic);
+
+  const { localConfigView } = useValues(ConnectorConfigurationLogic);
+  const { saveConfig, setIsEditing, setLocalConfigEntry } = useActions(ConnectorConfigurationLogic);
+
+  return (
+    <EuiForm
+      onSubmit={(event) => {
+        event.preventDefault();
+        saveConfig();
+      }}
+      component="form"
+    >
+      {localConfigView.map(({ key, label, value }) => (
+        <EuiFormRow label={label ?? ''} key={key}>
+          <EuiFieldText
+            value={value}
+            disabled={status === Status.LOADING}
```

Comment:
I'd extract all of these `status === Status.LOADING` checks into an `isLoading` variable, and also include Status.IDLE like

`const isLoading = status === Status.LOADING || status === Status.IDLE`

Reply:
Definitely don't want IDLE to mean disabled, because then you'll never be able to hit submit.

Reply:
This is based on the submit API status, not the fetch API status.